From df25ccc519ebd697796c1f07cc7485215169b498 Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Wed, 28 Nov 2018 23:56:31 +0100
Subject: [PATCH 12/23] tools: mktplinkfw2: add split-uboot layout

This commit adds the split-uboot partition layout used by the
Archer C50 v4 to mktplinkfw2.

Signed-off-by: David Bauer <mail@david-bauer.net>
---
 tools/firmware-utils/src/mktplinkfw2.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tools/firmware-utils/src/mktplinkfw2.c b/tools/firmware-utils/src/mktplinkfw2.c
index dead49e7af..35db848bba 100644
--- a/tools/firmware-utils/src/mktplinkfw2.c
+++ b/tools/firmware-utils/src/mktplinkfw2.c
@@ -146,6 +146,12 @@ static struct flash_layout layouts[] = {
 		.kernel_la	= 0x80000000,
 		.kernel_ep	= 0x80000000,
 		.rootfs_ofs	= 0x140000,
+	}, {
+		.id		= "8MSUmtk", /* Split U-Boot OS */
+		.fw_max_len	= 0x770000,
+		.kernel_la	= 0x80000000,
+		.kernel_ep	= 0x80000000,
+		.rootfs_ofs	= 0x140000,
 	}, {
 		.id		= "8MLmtk",
 		.fw_max_len	= 0x7b0000,
-- 
2.22.0

